event-sourcing/event-source-api/src/main/java/uk/gov/justice/services/eventsourcing/source/api/service/core/EventStreamService.java

```diff
@@ -19,52 +20,48 @@
-            final Direction direction,
-            final long pageSize) {
+    public List<EventStreamEntry> eventStreams(final Position position,
+                                               final Direction direction,
+                                               final long pageSize) {
         if (position.isHead()) {
-            return repository.head(pageSize).collect(toList());
+            final Stream<EventStream> eventStreamsFromFirst = eventSource.getEventStreamsBySequence(1);
+            final long sequenceNumber = eventStreamsFromFirst.count() - pageSize + 1;
+            final Stream<EventStreamEntry> eventStreamEntryStream = eventSource.getEventStreamsBySequence(sequenceNumber)
+                    .map(this::convertToEventStreamEntry);
+            final List<EventStreamEntry> eventStreamEntryList = eventStreamEntryStream.limit(pageSize).collect(toList());
+            return reverse(eventStreamEntryList);
         }
 
         if (position.isFirst()) {
-            return repository.first(pageSize).collect(toList());
+            final Stream<EventStream> eventStreams = eventSource.getEventStreamsBySequence(1);
+            return reverse(eventStreams.map(this::convertToEventStreamEntry).limit(pageSize).collect(toList()));
         }
 
         if (FORWARD.equals(direction)) {
-            return repository.forward(position.getSequenceId(), pageSize).collect(toList());
+            final Stream<EventStream> eventStreams = eventSource.getEventStreamsBySequence(position.getSequenceId());
+            return reverse(eventStreams.map(this::convertToEventStreamEntry).limit(pageSize).collect(toList()));
         }
 
         if (BACKWARD.equals(direction)) {
-            return repository.backward(position.getSequenceId(), pageSize).collect(toList());
+            final long sequenceNumber = position.getSequenceId() - pageSize + 1;
+            final Stream<EventStream> eventStreams = eventSource.getEventStreamsBySequence(sequenceNumber);
+            return reverse(eventStreams.map(this::convertToEventStreamEntry).limit(pageSize).collect(toList()));
         }
         return emptyList();
     }
 
-    public boolean recordExists(final long sequenceId) {
-        return repository.recordExists(sequenceId);
-    }
 
-    private List<EventStreamEntry> eventStreamEntries(final List<EventStream> events) {
-        return events.stream()
-                .map(toEventStreamEntry())
-                .collect(toList());
+    private EventStreamEntry convertToEventStreamEntry(final EventStream eventStream) {
+        return new EventStreamEntry(eventStream.getId().toString(), eventStream.getSequenceNumber());
     }
 
-    private Function<EventStream, EventStreamEntry> toEventStreamEntry() {
-        return eventStream -> new EventStreamEntry(
-                eventStream.getStreamId().toString(),
-                eventStream.getSequenceNumber()
-        );
+
+    public boolean recordExists(final long sequenceNumber) {
```

Comment:
recordExists?

eventExists ?
eventExistsAtPosition?

Reply:
Will this be retrieving all event streams? then running through each.
shouldn't we be doing this as a query?

Reply:
changed to eventStreamExists